### PR TITLE
Cleanup files created during tests

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

File: test/bin_test.py [hunk 1]
```diff
@@ -32,4 +38,11 @@ class LuigidDaemonTest(server_test.ServerTestRun):
 
     @mock.patch('daemon.DaemonContext')
     def run_server(self, daemon_context):
-        luigi.cmdline.luigid(['--port', str(self._api_port), '--background', '--logdir', '.', '--pidfile', self.id() + '.pid'])
+        tempdir = tempfile.mkdtemp(prefix=self.id())
```
Comment (erikbern) [inline test/bin_test.py]: hey you should probably use a context manager so you don't have to use `addCleanup`

https://docs.python.org/3/library/tempfile.html#examples

i've had some issues with tempfile before so not 100% that it works well

File: test/server_test.py [hunk 2]
```diff
@@ -95,6 +96,7 @@ def stop_server(self):
 
     def setUp(self):
         state_path = tempfile.mktemp(suffix=self.id())
+        self.addCleanup(functools.partial(os.unlink, state_path))
```
Comment (erikbern) [inline test/server_test.py]: same here

File: test/bin_test.py (hunk 1 above)
Comment (graingert) [inline test/bin_test.py]: It's difficult to use a context manager with unittest's method call interface, especially in this case where this is an overridden method

Comment (graingert) [inline test/bin_test.py]: Also python 2 doesn't have a tempdir context manager

Comment (erikbern) [inline test/bin_test.py]: ok got it – if it's not in python 2 then nvm!

**Comment by Tarrasch** [comment]
Sweeeeet

**Pull request merged**